File: 129-sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int res = 0;
    void dfs(TreeNode* root, int count)
    {
        if(root == NULL)
        {
            return;
        }
        count = (count*10) + root->val;
        if(!root->left && !root->right)
        {
            res+=count;
            return;
        }
        dfs(root->left, count);
        dfs(root->right, count);
    }
    int sumNumbers(TreeNode* root) {
        if(!root)
        {
            return 0;
        }
        dfs(root, 0);
        return res;
    }
};

File: 129-sum-root-to-leaf-numbers/sum-root-to-leaf-numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        s = 0
        def dfs(root, temp):
            nonlocal s
            if not root:
                return
            if not root.left and not root.right:
                s += (temp*10) + root.val
                return 
            temp = (temp * 10) + root.val
            dfs(root.left, temp)
            dfs(root.right, temp)
        dfs(root, 0)
        return s